### PR TITLE
chore: forward merge master onto the 'v2-main' branch

### PR DESCRIPTION
**Pull request by nija-at**
Create a merge PR three times a week (M/W/F) to forward merge master
onto 'v2-main'.

Mergify configuration to automatically merge the PR if there are no
conflicts and the sanity build succeeds.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: .github/workflows/v2-merge-forward.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: 'v2-main merge forward'
+on:
+  schedule:
+    - cron: '0 9 * * 1,3,5' # 9AM UTC every M/W/F
```
Comment (eladb) [inline .github/workflows/v2-merge-forward.yml:4]: Why not every day?

**Review by eladb — DISMISSED** [review]

File: .github/workflows/v2-merge-forward.yml (hunk 1 above)
Comment (nija-at) [reply .github/workflows/v2-merge-forward.yml:4]: Let's start here. Initially I want to see how this works. I'll increase frequency once everything is set up and I see smooth movement.

File: .github/workflows/v2-merge-forward.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        uses: actions/checkout@v2
+        with:
+          ref: 'v2-main'
+        run: |
+          git merge master
+      - name: Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          branch: v2/forward-merge
```
Comment (eladb) [inline .github/workflows/v2-merge-forward.yml:21]: Where is the part that creates a branch from v2 to v2/forward-merge?

Comment (nija-at) [reply .github/workflows/v2-merge-forward.yml:21]: As far as I can tell, here's how this works -

The `ref` at line 14 checks out the `v2-main` branch locally and later runs `git merge master` (line 16).

The `create-pull-request` GH action, takes care of pushing these new changes to a branch named `v2/forward-merge` and creates a pull request against the specified `base` branch.

From the [documentation](https://github.com/peter-evans/create-pull-request#-create-pull-request) -

>Create Pull Request action will:
>    1. Check for repository changes in the Actions workspace. This includes:
>        * untracked (new) files
>        * tracked (modified) files
>        * commits made during the workflow that have not been pushed
>    2. Commit all changes to a new branch, or update an existing pull request branch.
>    3. Create a pull request to merge the new branch into the base—the branch checked out in the workflow.

Comment (eladb) [reply .github/workflows/v2-merge-forward.yml:21]: Okay. Looks good

**Review by eladb — APPROVED** [review]

**Pull request merged**